src/ami-set.js

```diff
@@ -51,6 +55,49 @@ AmiSet.listAmiSets = async function () {
 };
 
 /**
+ * Checks if the AMIs from the amiSet are valid or not.
+ */
+
+async function checkAmi(ctx, region, ami, vtype) {
+  let request;
+  let missing = [];
+  let virtualizationType = '';
+
+  if (ami) {
```

Comment:
Hmm.  this is likely to make things silently fail.  I assume this is to check for the case that the hvm/pv thing is set to '' or not defined.  Let's instead not even call this function if the value is undefined.  In fact, let's have the first three lines of this function be assertions that the parameters passed in are as expected.  This makes debugging later easier.
```javascript
assert(typeof region === 'string');
assert(typeof ami === 'string');
assert(typeof vtype === 'string');
```